src/map/zone_entities.cpp

```diff
@@ -1367,6 +1367,29 @@ void CZoneEntities::ZoneServer(time_point tick, bool check_regions)
         it++;
     }
 
+    // Check to see if any aggroable mobs should be aggroed by other mobs
+    for (auto PMobIter : m_mobList)
+    {
+        CMobEntity* PMob = dynamic_cast<CMobEntity*>(PMobIter.second);
+        if (PMob == nullptr || PMob->allegiance != ALLEGIANCE_TYPE::PLAYER || !PMob->m_isAggroable)
+        {
+            continue;
+        }
+
+        for (auto PMobCurrentIter : m_mobList)
+        {
+            CMobEntity* PCurrentMob = dynamic_cast<CMobEntity*>(PMobCurrentIter.second);
+            if (PCurrentMob->isAlive() && PMob->allegiance != PCurrentMob->allegiance && distance(PMob->loc.p, PCurrentMob->loc.p) <= 50)
```

Comment:
Should be checking if this mob is null also, since the outer loop could have only iterated to 10, but the inner loop might be at 100

Reply:
Good catch!